lib/FusionInventory/Agent/Task/Inventory/Linux/Storages/Lsilogic.pm

```diff
@@ -18,7 +18,10 @@ sub doInventory {
     my $inventory = $params{inventory};
     my $logger    = $params{logger};
 
-    my @devices = getDevicesFromUdev(logger => $logger);
+    my @devices = map +{SCSI_UNID => $_}, map {/Found SCSI id=(\d+)/}
```

Comment:
Is there a good reason to have the plus sign (`+`) in `map +{SCSI_UNID => $_},` ?

Reply:
From the doc:
![2019-12-02-081433_950x141_escrotum](https://user-images.githubusercontent.com/3443972/69962349-d90b2100-14db-11ea-9666-e470151849d2.png)
The code may work without the plus sign, but i decided to make it by the book.

Reply:
Okay now I understand why you used such notation. I didn't know about it. The following notation would have been more clear (at least for me):
`my @devices = map { { SCSI_UNID => $_ } } map { /Found SCSI id=(\d+)/ }`

But then if a device entry is only including SCSI_UNID hash key, its NAME usage at l.28 is wrong. This just leaves the disk name as "undef". And this is probably not a big deal. But can you doubl-check this NAME key missing is not a problem on your side ?

After that, it would be nice to add a unit test with some `mpt-status -p` command outputs. You should even use a _getDevices() function where you explode the `my @devices = ...` into something more readable. Then you should be able to use: `my @devices = _getDevices( logger => $logger, command => 'mpt-status -p');` in doInventory(),
and `_getDevices(file => $output);` from test file.

Reply:
`@devices` is a list of controllers, not disks and `mpt-status` does not return names, only ids. I'm not sure what to do with `$device->{NAME}` now.

Will do _getDevices() and a test, although it sounds like an overkill.